docs/remote-access/wireguard.md

```diff
@@ -11,4 +11,112 @@ This page will cover two WireGuard architectures. The first is running WireGuard
+
+In the file above I have included the configuration for a second peer as an example on how to add multiple peers. 
+
+The final step is to start the server, and set it up to start automatically.
+
+```shell
+wg-quick up wg0
+sudo systemctl enable wg-quick@wg0
+```
+
+### Setup the Client
+```
+[Interface]
+# Client 1
+PrivateKey = << Private Key - client_1_key.priv >>
+ListenPort = 51902
+Address = 192.168.1.2/24
+
+PostUp = iptables -A FORWARD -i %i -j ACCEPT; iptables -A FORWARD -o %i -j ACCEPT; iptables -t nat -A POSTROUTING -o eth0 -j MASQUERADE
+PostDown = iptables -D FORWARD -i %i -j ACCEPT; iptables -D FORWARD -o %i -j ACCEPT; iptables -t nat -D POSTROUTING -o eth0 -j MASQUERADE
+
+
+[Peer]
+# Cloud
+PublicKey = << Public Key - cloud_key.pub >>
+PreSharedKey = << Preshared Key - client_1_cloud_preshared_key.psk >>
+Endpoint = wireguard.example.com:51902
+AllowedIPs = 192.168.1.0/24
+PersistentKeepalive = 25
+```
+
+The interface setup is the same as what we did on the server. Except we now need to use the client private key. 
+
+Now lets look at the Peer config for the connection to the cloud. The difference between the Server configuration and the client configuration is minor. The first is the endpoint is required, if you use a url instead of an IP address you need to be aware that if the IP changes you need to restart the wireguard connection, on the client. This is because the URL is converted to an IP when it is initiated. The final difference is the addition of the configuration PersistentKeepAlive. This is what allows us to use this VPN without any firewall openings. This sends data to the server to keep the connection alive. Without this wireguard is naturally a quiet connection, and only sends traffic when you do it intentionally. 
+
+# Test the connection. 
+
+To test the connection you first can test the connection within the wireguard configuration, from host client 1 send a ping like the following. 
+
+```shell
+ping 192.168.1.1
+```
+
+If all is working this should look like this:
+```
+PING 192.168.1.1 (192.168.1.1) 56(84) bytes of data.
+64 bytes from 192.168.1.1: icmp_seq=1 ttl=64 time=26.8 ms
+64 bytes from 192.168.1.1: icmp_seq=2 ttl=64 time=26.7 ms
+64 bytes from 192.168.1.1: icmp_seq=3 ttl=64 time=28.1 ms
+--- 192.168.1.1 ping statistics ---
+6 packets transmitted, 6 received, 0% packet loss, time 5009ms
+rtt min/avg/max/mdev = 26.745/28.130/31.222/1.498 ms
+```
+
```

Comment:
At this point whilst connections from peers to the host will work, transport between peers won't.

You'll need to add `sysctl -w net.ipv4.ip_forward=1` to enable that to work. I think this is definitely worth talking about here.

Reply:
I forgot about this. I will add this to the documentation as well.

Reply:
I added this in my second draft.